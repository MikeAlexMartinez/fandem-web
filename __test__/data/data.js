const casual = require("casual");

casual.seed(777);

const fakeUser = () => ({
  __typename: "User",
  id: casual.uuid,
  displayName: casual.word,
  name: casual.full_name,
  phoneNumber: casual.phone,
  email: casual.email,
  userRoles: [
    {
      __typename: "UserRole",
      name: "USER"
    }
  ],
  followers: {
    __typename: "UserFollower",
    followers: []
  },
  country: null,
  profilePicture: [],
  influencers: {
    __typename: "UserInfluencer",
    influencers: []
  },
  countryCode: null,
  isPrivate: false,
  subscriptions: [
    {
      __typename: "Subscription",
      name: "FREE"
    }
  ],
  emailValidated: false,
  favoriteTeam: null
});

const fakeFavoriteTeam = () => ({
  name: "Spurs",
  shortName: "TOT"
});

export { fakeUser, fakeFavoriteTeam };
