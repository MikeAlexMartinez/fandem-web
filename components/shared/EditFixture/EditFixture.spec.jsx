import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import EditFixtureComponent from './EditFixture';

import { fakeFixture } from '../../../__test__/data/data';

describe('<EditFixture />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(
      <EditFixtureComponent
        open
        fixture={fakeFixture()}
        handleClose={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
