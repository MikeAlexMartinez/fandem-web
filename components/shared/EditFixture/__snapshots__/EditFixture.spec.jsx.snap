// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditFixture /> renders and matches the snapshot 1`] = `
<EditFixture
  classes={
    Object {
      "actionRoot": "EditFixture-actionRoot-5",
      "actions": "EditFixture-actions-4",
      "container": "EditFixture-container-1",
      "formGrid": "EditFixture-formGrid-2",
      "input": "EditFixture-input-3",
    }
  }
  fixture={
    Object {
      "__typename": "Fixture",
      "event": Object {
        "id": "cjyjtpb940075079888jpdt4y",
        "name": "Gameweek 1",
      },
      "finished": false,
      "fixtureId": 1,
      "fplCode": 1059702,
      "id": "cjycw2ups0i3s0787mh7ya6x6",
      "kickoffTime": "2019-08-09T19:00:00.000Z",
      "minutes": 0,
      "teamA": Object {
        "__typename": "AwayTeamFixture",
        "awayTeam": Object {
          "__typename": "Team",
          "name": "Norwich",
        },
      },
      "teamAScore": null,
      "teamH": Object {
        "__typename": "HomeTeamFixture",
        "homeTeam": Object {
          "__typename": "Team",
          "name": "Liverpool",
        },
      },
      "teamHScore": null,
    }
  }
  handleClose={[Function]}
  open={true}
/>
`;
